### PR TITLE
[4341][next] Update Wordify BP to make ready for MP

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4341

### CODE REVIEW

File: sources/react-app/public/index.html [hunk 1]
```diff
@@ -103,6 +103,7 @@
     <link rel="stylesheet" href="/static-assets/css/style.css">
 
     <title>Wordify Crafter CMS</title>
+    <script id="2fb5164e" type="application/json">${siteName!'null'}</script>
```
Comment (rart) [inline sources/react-app/public/index.html:106]: not json

File: sources/react-app/src/pages/Category.js [hunk 2]
```diff
@@ -123,9 +123,16 @@ export default function (props) {
...
-  const categoryId = match.params.id;
+  const [categoryId, setCategoryId] = useState();
   let resource = useTaxonomiesResource();
 
+
+  useEffect(() => {
+    if (match.path === '/category/:id?' || match.path === '/tag/:id?') {
+      setCategoryId(match.params.id);
+    }
+  }, [match.params, match.path]);
```
Comment (rart) [inline sources/react-app/src/pages/Category.js:134]: Why is the route something other than this at times?

File: config/studio/context-nav/sidebar.xml [hunk 3]
```diff
@@ -31,97 +31,121 @@
...
-                    <path>/site-config</path>
-                  </params>
+                    <locale>en</locale>
+                  </props>
+                  <container>
+                    <style>
+                      <padding-left>9px</padding-left>
+                    </style>
+                    <class>next-pages-widget</class>
+                  </container>
```
Comment (rart) [inline config/studio/context-nav/sidebar.xml:148]: x

File: config/studio/context-nav/sidebar.xml [hunk 4]
```diff
@@ -31,97 +31,121 @@
...
-                    <onClick>none</onClick>
-                  </params>
+                    <locale>en</locale>
+                  </props>
+                  <container>
+                    <style>
+                      <padding-left>9px</padding-left>
+                    </style>
+                    <class>next-pages-widget</class>
+                  </container>
```
Comment (rart) [inline config/studio/context-nav/sidebar.xml:131]: x

File: sources/react-app/src/shared/DynamicRoute.js [hunk 5]
```diff
@@ -91,7 +91,10 @@ export default function DynamicRoute(props) {
     return <CircularProgressSpinner />;
   } else {
     return (
-      <Guest modelId={state.model?.craftercms.id} isAuthoring={isAuthoring()}>
+      <Guest
+        modelId={state.model?.craftercms.id} isAuthoring={isAuthoring()}
+        path={state.model?.craftercms.path}
```
Comment (rart) [inline sources/react-app/src/shared/DynamicRoute.js:96]: Break all 3 onto their own line

File: sources/react-app/src/shared/utils.js [hunk 6]
```diff
@@ -53,7 +53,9 @@ export function createResource(factory) {
   };
 }
 
-export const siteName = process.env.REACT_APP_CRAFTERCMS_SITE_ID || Cookies.get('crafterSite');
+export const siteName = document.getElementById('2fb5164e').innerHTML ||
+  process.env.REACT_APP_CRAFTERCMS_SITE_ID ||
+  Cookies.get('crafterSite');
```
Comment (rart) [inline sources/react-app/src/shared/utils.js:58]:
```js
export const siteName = 
    document.getElementById('2fb5164e')?.innerHTML ||
    process.env.REACT_APP_CRAFTERCMS_SITE_ID ||
    Cookies.get('crafterSite');
```

**Review by rart — CHANGES_REQUESTED** [review]

File: sources/react-app/src/pages/Category.js (hunk 2 above)
Comment (jvega190) [reply sources/react-app/src/pages/Category.js:134]: Actually it isn't. Removing validation

**Review by rart — CHANGES_REQUESTED** [review]

File: sources/react-app/src/shared/DynamicRoute.js [hunk 7]
```diff
@@ -31,10 +31,14 @@ const limit = 3;
...
   const { match, location } = props;
   const [state, setState] = useState(null);
-  const url = match.path.includes(':')
+  let url = match.path.includes(':')
     ? match.path.substring(0, match.path.indexOf(':') -1)
     : match.url;
 
+  if (url.includes('/post')) {
+    url += '.xml';
+  }
```
Comment (rart) [inline sources/react-app/src/shared/DynamicRoute.js:40]: What's this for? This looks unnecessary.

**Review by rart — CHANGES_REQUESTED** [review]

File: config/engine/urlrewrite.xml [hunk 8]
```diff
@@ -16,12 +15,9 @@
   ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   -->
 
-<!DOCTYPE urlrewrite
-  PUBLIC "-//tuckey.org//DTD UrlRewrite 4.0//EN"
-  "http://www.tuckey.org/res/dtds/urlrewrite4.0.dtd">
```
[the remark this answers is not in the document]
Comment (jvega190) [reply config/engine/urlrewrite.xml:21]: Commented - waiting for fix in https://github.com/craftercms/craftercms/issues/4380

File: sources/react-app/.env.development [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 # Please specify the base URL where Crafter CMS is running
 # This is used for configuring the sdk requests
-REACT_APP_CRAFTERCMS_BASE_URL=//localhost:8080
+REACT_APP_CRAFTERCMS_BASE_URL=//localhost:3000
```
Comment (rart) [inline sources/react-app/.env.development:5]: ?

**Review by rart — CHANGES_REQUESTED** [review]

File: sources/react-app/.env.development (hunk 9 above)
Comment (jvega190) [reply sources/react-app/.env.development:5]: reverting, pushed the file by mistake

**Review by rart — CHANGES_REQUESTED** [review]
Singularize bios dir on components

**Pull request merged**